src/asciidoc/reference/packages/withkojiDatabase.adoc

```diff
@@ -62,28 +151,145 @@ const myData = await database.get('myCollection');
 const myValue = await database.get('myCollection','myKey');
 ----
 
+[.hcode, id=".getAll", reftext="getAll"]
+=== .getAll(collection, keys)
+
+Gets all the specified database entries.
+
+==== Parameters
+
+* `collection` – _String_, name of the collection.
+* `keys` – Array of _Strings_, list of keys of the database entries to retrieve.
+
+==== Returns
+
+(Async) Array of _Objects_, the requested database entries from the collection.
+
+==== Example
+
+[source, javascript]
+const myValue = await database.getAll('myCollection',['myKey1', 'myKey2']);
+
+[.hcode, id=".getAllWhere", reftext="getAllWhere"]
+=== .getAllWhere(collection, predicateKey, predicateOperation, predicateValues)
+
+Gets all the database entries that match a query against a list of possible values.
+
+==== Parameters
+
+* `collection` – _String_, name of the collection.
+* `predicateKey` – _String_, key of the data to query against.
+* `predicateOperation` – _String_, operation to query the data against, possible values are `<`, `\<=`, `==`, `>`, `>=`.
+* `predicateValues` – Array of _Strings_, list of values to run the query against.
+
+==== Returns
+
+(Async) Array of _Objects_, database entries that match the queries.
+
+==== Example
+
+[source, javascript]
+const results = await database.getAllWhere('myCollection','predicateKey', '==', ['predicateValue1', 'predicateValue2']);
+
+[.hcode, id=".getCollections", reftext="getCollections"]
+=== .getCollections()
+
+Gets all the collections available in the database.
+
+==== Returns
+
+(Async) Array of _Strings_, list containing the names of the collections.
+
+==== Example
+
+[source, javascript]
+const collections = await database.getCollections();
+
+[.hcode, id=".getWhere", reftext="getWhere"]
+=== .getWhere(collection, predicateKey, predicateOperation, predicateValue)
+
+Gets all the database entries that match a query against a value.
+
+==== Parameters
+
+* `collection` – _String_, name of the collection.
+* `predicateKey` – _String_, key of the data to query against.
+* `predicateOperation` – _String_, operation to query the data against, possible values are `<`, `\<=`, `==`, `>`, `>=`.
+* `predicateValue` – _String_, value to run the query against.
+
+==== Returns
+
+(Async) Array of _Objects_, list of the database entries that match the query.
+
+==== Example
+
+[source, javascript]
+const results = await database.getWhere('myCollection','predicateKey', '==', 'predicateValue');
+
+[.hcode, id=".search", reftext="search"]
+=== .search(collection, queryKey, queryValue)
+
+Gets all the database entries that have a partial match on a particular key
+
+NOTE: The `queryKey` must be set for all entries in your data, if some entries have no value for that key it cannot be searched on.
```

Comment:
Perhaps wrote this in a misleading way. It doesn't return an error, it just doesn't return anything.
This may be a bug in the package (or the underlying database service) as it seems like odd functionality.